core/java/android/ext/settings/ClipboardConstants.java

```diff
@@ -0,0 +1,7 @@
+package android.ext.settings;
+
+/** @hide */
+public class ClipboardConstants {
+    // From com.android.server.clipboard.ClipboardService.DEFAULT_CLIPBOARD_TIMEOUT_MILLIS 
+    public static final int DEFAULT_CLIPBOARD_TIMEOUT_MILLIS = 3600000;
```

Comment:
There's no need to declare this is a separate constant in a separate file.
It's better to use `TimeUnit.HOURS.toMillis(1)` in `CLIPBOARD_AUTO_CLEAR_TIMEOUT` definition (default value can likely be reduced to several minutes) and to remove constant `com.android.server.clipboard.ClipboardService.DEFAULT_CLIPBOARD_TIMEOUT_MILLIS` to make sure it's not used elsewhere.

Reply:
This is just following the convention from `com.android.server.clipboard.ClipboardService.DEFAULT_CLIPBOARD_TIMEOUT_MILLIS` for making it easier to port to other versions of AOSP.

Reply:
This also includes changing CTS as this value is used there. Depends if you don't mind.

Reply:
If constant is removed, build will fail if it's used elsewhere in a new AOSP version. This is more reliable than assuming that search for usages of this constant will be performed each time new AOSP version comes out.

Reply:
Why would you want to use DeviceConfig? Isn't the point of ExtSettings is to easily add new settings? This constant (`ClipboardConstants.DEFAULT_CLIPBOARD_TIMEOUT_MILLIS`) is not used elsewhere, only in `ExtSettings`, thus making it easier to port.

Reply:
I'm referring to `com.android.server.clipboard.ClipboardService.DEFAULT_CLIPBOARD_TIMEOUT_MILLIS` constant, not to `ClipboardConstants.DEFAULT_CLIPBOARD_TIMEOUT_MILLIS`

Reply:
Why add a new setting when AOSP already has a setting for the same thing?

Reply:
Oh yes but we tried to use it and it can't be imported (`ClipboardService.DEFAULT_CLIPBOARD_TIMEOUT_MILLIS`). You mean adding it to `DeviceConfig` as a setting?

Reply:
I mean that the setting is already present in AOSP, it just doesn't have a UI:
https://github.com/GrapheneOS/platform_frameworks_base/blob/71a9bb51861736106919e97afec14a9bccdc4e10/services/core/java/com/android/server/clipboard/ClipboardService.java#L432

Reply:
I understand. Then instead of using `ExtSettings.CLIPBOARD_AUTO_CLEAR_TIMEOUT` in the [controller](https://github.com/GrapheneOS/platform_packages_apps_Settings/pull/158/commits/70dd2301f20507fe30154e035b958d890747fac2) we would use `DeviceConfig`. But we would need now to implement `AbstractListPreferenceController` because `IntSettingPrefController` uses `IntSetting`.

Reply:
Using `DeviceConfig` can be a bit problematic since we need to use some members from `ClipboardService` in [`ClipboardTimeoutController`](https://github.com/GrapheneOS/platform_packages_apps_Settings/pull/158/commits/70dd2301f20507fe30154e035b958d890747fac2) (and `com.android.server.clipboard` can't be imported.) A fixable solution is to use `ClipboardManager` instead but we would need to add a few methods that use the (needed) members from `ClipboardService`.

Reply:
Using `DeviceConfig` requires `PROPERTY_AUTO_CLEAR_TIMEOUT`, and `DEFAULT_CLIPBOARD_TIMEOUT_MILLIS` (if we want consistency) from `ClipboardService`. It's possible with a few changes.

Reply:
Can just move these fields to ClipboardManager.

Reply:
How would we make it per-user? The current implementation doesn't allow per-user configuration because it uses `DeviceConfig`.

Reply:
I don't think this setting needs to be per-user.

Reply:
It adds better isolation, and even security (no leaks between user profiles).

Reply:
DeviceConfig settings are unreadable by unprivileged apps.

I've thought some more about this feature, and I think that it's better to just reduce this timeout to 10 or 5 minutes instead of adding one more setting. There's a large number of privacy-sensitive timeouts and flags in AOSP, making them configurable is not a right approach in most cases in my opinion.

Reply:
I believe it's still a good feature to add. People can copy passwords (for some reason) or other sensitive information from other apps and not worry about it existing for a long time and some app stealing it. It's also a pretty small change. Just a "setting" file and copying two constants. Can be easily added.